File: BE/Src/MIT.Fwk.Infrastructure/Entities/Category.cs
using JsonApiDotNetCore.Resources;
using JsonApiDotNetCore.Resources.Annotations;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace MIT.Fwk.Infrastructure.Entities
{
    [Table("Category")]
    public class Category : Identifiable<int>
    {

        [Attr] public string Name { get; set; }
        [Attr] public string Description { get; set; }
        [Attr] public string Type { get; set; }
        [Attr] public int ParentCategory { get; set; }
        [Attr] public bool Erasable { get; set; }
        [Attr] public bool CopyInNewTenants { get; set; }
        [Attr] public int TenantId { get; set; }
        [Attr] public int Order { get; set; }
        [Attr] public string Code { get; set; }

        [HasOne]
        public virtual Tenant Tenant { get; set; }

        [HasMany]
        public virtual ICollection<Template> Template { get; set; }



    }
}

File: FE/WEB/app/models/category.js
import Model, { attr, hasMany, belongsTo } from '@ember-data/model';

export default class CategoryModel extends Model {
  @hasMany('template', { async: true, inverse: 'category' }) template;
  @belongsTo('tenant', { async: true, inverse: 'category' }) tenant;

  @attr('string', { defaultValue: '' }) name;
  @attr('string', { defaultValue: '' }) code;
  @attr('string', { defaultValue: '' }) description;
  @attr('string', { defaultValue: '' }) type;
  @attr('number', { defaultValue: 0 }) parentCategory;
  @attr('number', { defaultValue: '' }) tenantId;
  @attr('boolean', { defaultValue: false }) copyInNewTenants; // indica se la categoria e i suoi template vanno copiati nei nuovi tenant

  @attr('number', { defaultValue: 0 }) order;
  @attr('boolean', { defaultValue: true }) erasable;
}
